lab_1_keywords_tfidf/main.py

```diff
@@ -2,158 +2,256 @@
 Lab 1
 Extract keywords based on frequency related metrics
 """
-from typing import Optional, Union
+import math
+from typing import Optional, Union, Any
+import string
 
 
-def clean_and_tokenize(text: str) -> Optional[list[str]]:
+def correct_list(list1: list, type1: Any, empty: bool) -> bool:
     """
-    Removes punctuation, casts to lowercase, splits into tokens
-
-    Parameters:
-    text (str): Original text
-
-    Returns:
-    list[str]: A sequence of lowercase tokens with no punctuation
-
-    In case of corrupt input arguments, None is returned
+    Checks that type of 'list1' is list and verifies the contents of 'list1' with the type that the user specifies
     """
-    pass
+    if isinstance(list1, list):
+        if list1 == [] and empty is False:
```

Comment:
simplify the condition. You don't need to directly compare with `False` and an empty list. https://www.freecodecamp.org/news/truthy-and-falsy-values-in-python/

Reply:
Done